client/allocdir/fs_darwin.go

```diff
@@ -24,3 +24,14 @@ func createSecretDir(dir string) error {
 func removeSecretDir(dir string) error {
 	return os.RemoveAll(dir)
 }
+
+// bindMount mounts src to dst with support for read only mounting.
+// Assumes that filepath.Dir(dst) exists already
+func bindMount(src, dst string, readOnly bool) error {
+	return syscall.Link(src, dst)
```

Comment:
I added mount functions to `fs` of the unix paths for completion sake and they are similar to `linkDir` - However, I don't believe the linux fs_linux.go operations are the same as the rest of unix ones, e.g. `linkDir` in linux creates the mount points, while others don't - so I suspect we have some untested paths here.  Not sure where to note these though.